ui/src/app/store/scriptresult/types.ts

```diff
@@ -56,7 +56,39 @@ export type ScriptResult = PartialScriptResult & {
   hardware_type: HardwareType;
   interface?: NetworkInterface | null;
   name: string;
-  parameters?: Record<string, unknown>;
+  parameters?: {
+    interface?: {
+      type: "interface";
```

Comment:
It might be nice to use an enum for these interface types so you don't have to use the correct string in places like line 116 of MachineTests.test.tsx